std/assembly/array.ts

```diff
@@ -179,6 +179,29 @@ export class Array<T> {
     return newLength;
   }
 
+  concat(items: Array<T>): Array<T> {
+    var thisLen: isize = this.length_;
+    var otherLen = (items == null) ? 0 : items.length_;
+    var outLen =  thisLen + otherLen;
+    var out: Array<T> = new Array<T>(outLen);
+
+    if (thisLen) {
+      memory.copy(changetype<usize>(out.buffer_) + HEADER_SIZE,
+      changetype<usize>(this.buffer_) + HEADER_SIZE,
+      <usize>(thisLen << alignof<T>()));
+
+      if (isManaged<T>()) __gc_link(changetype<usize>(out), changetype<usize>(this)); // tslint:disable-line
```

Comment:
Why is it registering `this` array as a child (element) of the `out` array here?

Reply:
btw how handle managed references which is array's elements in this case? I also wonder how that would be right?

Reply:
@LiaoPeng remove plz `__gc_link`

Reply:
@MaxGraey OK.

Reply:
Second one also. @dcodeIO right?

Reply:
Yup

Reply:
Second one also